flytekit/common/launch_plan.py

```diff
@@ -66,6 +72,29 @@ def promote_from_model(cls, model):
             raw_output_data_config=model.raw_output_data_config,
         )
 
+    @_exception_scopes.system_entry_point
+    def register(self, project, domain, name, version):
```

Comment:
can we change register to take in a context variable, maybe a separate PR. I want to use this context to specify endpoints, image etc as inputs instead of automatically figuring it out

Reply:
yup, next PR.